tests/test_external_search.py

```diff
@@ -396,490 +453,275 @@ def test_query_works(self):
+        # Search in subtitle
+        expect(self.moby_dick, "whale")
 
-        # Matches audience
+        # Search in series.
+        expect(self.moby_dick, "classics")
 
-        results = query("children's", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(1, len(hits))
-        eq_(unicode(self.children_work.id), hits[0]['_id'])
+        # Search in summary.
+        expect(self.moby_dick, "ishmael")
 
-        results = query("young adult", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
-        work_ids = sorted([unicode(self.ya_work.id), unicode(self.ya_romance.id)])
-        result_ids = sorted([hit["_id"] for hit in hits])
-        eq_(work_ids, result_ids)
+        # Search in publisher name.
+        expect(self.moby_dick, "gutenberg")
 
+        # Title > subtitle > summary > publisher.
+        expect(
+            [self.title_match, self.subtitle_match,
+             self.summary_match, self.publisher_match],
+            "match"
+        )
 
-        # Matches grade
+        # (title match + author match) > title match
+        expect(
+            [self.moby_dick, self.moby_duck],
+            "moby melville"
+        )
 
-        results = query("grade 4", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
-        eq_(unicode(self.age_9_10.id), hits[0]['_id'])
+        # Match a quoted phrase
+        # 'Moby-Dick' is the first result because it's an exact title
+        # match. 'Moby Duck' is the second result because it's a fuzzy
+        # match,
+        expect([self.moby_dick, self.moby_duck], '"moby dick"')
 
-        results = query("grade 4-6", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
-        eq_(unicode(self.age_9_10.id), hits[0]['_id'])
+        # Match a stemmed word: 'running' is stemmed to 'run', and
+        # so is 'runs'.
+        expect(self.dodger, "runs")
 
+        # Match a misspelled phrase: 'movy' -> 'moby'.
+        expect([self.moby_dick, self.moby_duck], "movy", ordered=False)
 
-        # Matches age
+        # Match a misspelled author: 'mleville' -> 'melville'
+        expect(self.moby_dick, "mleville")
 
-        results = query("age 9", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
-        eq_(unicode(self.age_9_10.id), hits[0]['_id'])
+        expect([self.moby_dick, self.moby_duck], "mo by dick")
 
-        results = query("age 10-12", None, None, None, None, None, None, None)
-        hits = results["hits"]["hits"]
-        eq_(2, len(hits))
-        eq_(unicode(self.age_9_10.id), hits[0]['_id'])
+        # A query without an apostrophe matches a word that contains one.
+        # (NOTE: it's not clear whether this is a feature of the index or
+        # done by the fuzzy match.)
```

Comment:
I can't remember for sure if the stemmer does this, but it's likely handled by both.